README.md

```diff
@@ -1 +1,161 @@
-# node-modules-path
\ No newline at end of file
+# Paths, Folders and Modules
+
+## Overview
+
+This lesson will cover how to work with paths, use them when importing modules. We'll also show a code organization technique where you can import folders.
+
+## Objectives
+
+1. Describe `__dirname`, `process.cwd()` and `process.env.PWD`
+1. Describe how to import folders (and why)
+
+
+## Absolute Path with __dirname
+
+Consider the code in which we read a file on Mac OS X. The file is in a `data` folder relative to our script:
+
+```js
+var fs = require('fs')
+var customers = fs.readFileSync('data/customers.csv', {encoding: 'utf8'})
+console.log(customers)
+```
+
+Unfortunately, this code will break on Windows. This is due to a different path syntax, i.e., `data\customers.csv`. That's why we use `path.join()` which we discussed in one of the previous lessons. This example is in the file `customers.js`:
+
+```js
+var fs = require('fs')
+var path = require('path')
+var customers = fs.readFileSync(path.join('data','customers.csv'), {encoding: 'utf8'})
+console.log(customers)
+```
+
+However, if we attempt to run the `customers.js` file from a patent folder with `$ node node-modules-path/customers.js`, the code will produce an error:
+
+```
+Error: ENOENT: no such file or directory, open 'data/customers.csv'
+```
+
+The error tells us that there's no such directory and in fact that's true. Our data folder is in `node-modules-path`, and not in its parent folder. One option is to edit the file to reflect the change:
+
+```
+var fs = require('fs')
+var path = require('path')
+var customers = fs.readFileSync(path.join('node-modules-path', 'data', 'customers.csv'), {encoding: 'utf8'})
+console.log(customers)
+```
+
+The script will work from the parent folder now, but I think you see the problem with this approach. Every time we change the folder from which we launch our Node script, we have to modify the source code. This is not a good way to write programs. A better way is to use absolute path, but how do we get it? 
+
+To get an absolute path, there are a method `process.cwd()`, and two properties `__dirname` `process.env.PWD`. They a slightly different. We start with the property `__dirname`, because that's what we need to solve this problem. 
+
+`__dirname` returns the global path to the location of the code being executed. Our problem is fixed by changing relative path to the CSV file to an absolute one:
+
+```
+var fs = require('fs')
+var path = require('path')
+var customers = fs.readFileSync(path.join(__dirname, 'data', 'customers.csv'), {encoding: 'utf8'})
+console.log(customers)
+```
+
+Yay! Try to run this code on your own from different locations like:
+
+```
+$ node customers.js
+$ node node-modules-path/customers.js
+```
```

Comment:
aren't there problems with absolute paths once you deploy this app? the absolute path on your computer will be very different from the production absolute path